reactor-netty-http/src/test/java/reactor/netty/http/Http2Tests.java

```diff
@@ -491,29 +506,57 @@ void testIssue1394_HttpEndpointH2Protocol() {
 
 		SslContextBuilder clientCtx = SslContextBuilder.forClient();
 		HttpClient.create()
-		          .protocol(HttpProtocol.HTTP11, HttpProtocol.H2)
+		          .protocol(protocols)
 		          .secure(spec -> spec.sslContext(clientCtx))
 		          .wiretap(true)
 		          .get()
 		          .uri("http://localhost:" + disposableServer.port() + "/")
-		          .responseSingle((res, bytes) -> Mono.just(res.responseHeaders().get("x-http2-stream-id")))
+		          .responseSingle((res, bytes) -> Mono.just(res.responseHeaders().get("x-http2-stream-id", "null")))
 		          .as(StepVerifier::create)
-		          .expectNext("1")
+		          .expectNext(expectedStreamId)
 		          .expectComplete()
 		          .verify(Duration.ofSeconds(30));
 	}
 
 	@Test
-	void testIssue1394_HttpsEndpointH2CProtocol() {
+	void testIssue1394__SchemeHttps() {
+		doTestIssue1394__SchemeHttps(s -> !"null".equals(s), HttpProtocol.H2);
+		doTestIssue1394__SchemeHttps(s -> !"null".equals(s), HttpProtocol.H2, HttpProtocol.H2C);
+		doTestIssue1394__SchemeHttps(s -> !"null".equals(s), HttpProtocol.HTTP11, HttpProtocol.H2, HttpProtocol.H2C);
+		doTestIssue1394__SchemeHttps(s -> !"null".equals(s), HttpProtocol.HTTP11, HttpProtocol.H2);
+
+		doTestIssue1394__SchemeHttps("null"::equals, HttpProtocol.HTTP11);
+		doTestIssue1394__SchemeHttps("null"::equals, HttpProtocol.HTTP11, HttpProtocol.H2C);
+	}
+
+	private void doTestIssue1394__SchemeHttps(Predicate<String> predicate, HttpProtocol... protocols) {
 		HttpClient.create()
-		          .protocol(HttpProtocol.HTTP11, HttpProtocol.H2C)
+		          .protocol(protocols)
 		          .wiretap(true)
 		          .get()
 		          .uri("https://example.com")
-		          .responseSingle((res, bytes) -> Mono.just(res.responseHeaders().get("x-http2-stream-id")))
+		          .responseSingle((res, bytes) -> Mono.just(res.responseHeaders().get("x-http2-stream-id", "null")))
 		          .as(StepVerifier::create)
-		          .expectNext("3")
+		          .expectNextMatches(predicate)
 		          .expectComplete()
 		          .verify(Duration.ofSeconds(30));
 	}
+
+	@Test
+	void testIssue1394_ProtocolSchemeNotCompatible() {
+		doTestIssue1394_ProtocolSchemeNotCompatible(HttpProtocol.H2, "http");
+		doTestIssue1394_ProtocolSchemeNotCompatible(HttpProtocol.H2C, "https");
+	}
+
+	private void doTestIssue1394_ProtocolSchemeNotCompatible(HttpProtocol protocol, String scheme) {
+		HttpClient.create()
+		          .protocol(protocol)
+		          .wiretap(true)
+		          .get()
+		          .uri(scheme + "://example.com")
+		          .responseSingle((res, bytes) -> Mono.just(res.responseHeaders().get("x-http2-stream-id")))
+		          .as(StepVerifier::create)
+		          .expectError(IllegalArgumentException.class)
```

Comment:
having a meaningful message would be beneficial, and testing it here would be a form of minimal documentation

Reply:
done